### PR TITLE
Refactor bch-toolkit

### PR DESCRIPTION
**Pull request by nicolaiskye**
### Changes
* The setup script has been simplified, and is able to set up environments quicker than the previous design of bch-toolkit could.
* I have refactored bch-toolkit to be more flexible, in that you can more easily swap out components (node, indexer, etc) by simply changing which docker-compose file it points to (located in the new dc_files directory). This is something that will most likely be auto-generated in future versions of the scripts when I switch them over to Python.
* Docker images are no longer built in bch-toolkit .. instead electing to host them on Docker Hub under the ActorForth organization there. This should speed things up by not requiring each user to build their own containers when downloading and using the bch-toolkit. 
* 'run' script changed to 'services' script, which now has the ability to both start and stop the docker containers.
* Configuration files now have their own directory
* Git sub-modules removed
* Documentation updated to reflect latest updates and usage

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -4,49 +4,109 @@ services:
         image: zquestz/bitcoin-unlimited
         container_name: bitcoind-regtest
         ports:
-          - "18332:18332"
+          - "18443:18443"
```
Comment (SuperCipher) [inline docker-compose.yml:7]: I just test setup and run and it seems ./bitcoin-cli can't connect to server
Screen Shot 2021-01-04 at 12 58 27

File: docker-compose.yml [hunk 2]
```diff
@@ -4,49 +4,109 @@ services:
...
+          - ./configs/bitcoin-unlimited.conf:/root/.bitcoin/bitcoin.conf
         entrypoint: "/entrypoint.sh"
         command: bitcoind
         networks:
           - bitcoin-bridging
 
-    nginx-regtest:
-        image: nginx:latest
-        container_name: nginx-regtest
+    # bitcoind-regtest:
```
Comment (SuperCipher) [inline docker-compose.yml:19]: Duplication.

File: docker-compose.yml [hunk 3]
```diff
@@ -4,49 +4,109 @@ services:
...
+    #     volumes:
+    #       - ./data/bitcoind:/data
+    #       - ./configs/bitcoin.conf:/data/bitcoin.conf
+    #       - ./configs/bitcoin.conf:/root/.bitcoin/bitcoin.conf
+    #     entrypoint: "/entrypoint.sh"
+    #     command: bitcoind
+    #     networks:
+    #       - bitcoin-bridging
+
+    # fulcrum:
```
Comment (SuperCipher) [inline docker-compose.yml:34]: What is this do?

File: setup [hunk 4]
```diff
@@ -55,81 +81,145 @@ else
...
+if [[ $BTK_REST == 1 ]]; then
+	# export COMPOSE_FILE="$COMPOSE_FILE:./dc_files/docker-compose.opensight.yml"
+	# export COMPOSE_FILE="$COMPOSE_FILE:./dc_files/docker-compose.restapi.yml"
+	docker pull actorforth/opensight:latest
+	docker pull actorforth/rest.bitcoin.com:latest
+fi
 
-	#
-	touch .BTK-container-lock
+if [[ $BTK_JS_API == 1 ]]; then
```
Comment (SuperCipher) [inline setup:184]: I test the ./setup script and there is not setup bch-api.

Comment (nicolaiskye) [reply setup:184]: Yeah, I have an idea for a workaround to make bch-api an option again. I'll add that as an additional commit once I figure something out.

File: docker-compose.yml (hunk 2 above)
Comment (nicolaiskye) [reply docker-compose.yml:19]: They're not duplicated. The second (commented out) one is for Bitcoin Cash Node and Fulcrum indexer, whereas the first uncommented on is for Bitcoin Unlimited (and its included Electrs indexer). Currently BCHN and Fulcrum have a weird bug that causes them to reboot into 'mainnet' mode. I'm just leaving the comments here so I can continue to work on and debug it so we'll have support in the future for multiple nodes and indexers.

File: docker-compose.yml (hunk 1 above)
Comment (nicolaiskye) [reply docker-compose.yml:7]: Strange, although I don't think it's related to the port being changed. Technically 18332 is meant for testnet, and 18443 is meant for regtest. I'll test things out a bit more and see if I can reproduce this issue.

File: docker-compose.yml (hunk 3 above)
Comment (nicolaiskye) [reply docker-compose.yml:34]: This is the Bitcoin Cash Node and Fulcrum indexer (as opposed to the Bitcoin Unlimited node that we currently use). Unfortunately it has the "mainnet" bug currently and I'm still figuring that out. Continue using BU for now.

**Pull request merged**